### ISSUE TITLE
Missing logs when parsing nested json

### ISSUE THREAD

**Issue by alexanderccc** [fluent/fluent-bit#2628]
## Bug Report

**Describe the bug**
I'm using fluent-bit to parse the logs from Kubernetes micro services and send them to es. By default I have the tail input with the docker parser and the Kubernetes Filter. If I add an additional filter to parse the nested "message" field some messages randomly disappear (are not being sent to elastic)

**Your Environment**
* Version used: 1.5.7
* Environment name and version (e.g. Kubernetes? What version?): v1.15.11

This is the configuration that works as expected
```
    [FILTER]
        Name                kubernetes
        Match               kube.*
        Kube_URL            https://kubernetes.default.svc:443
        Kube_CA_File        /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
        Kube_Token_File     /var/run/secrets/kubernetes.io/serviceaccount/token
        Kube_Tag_Prefix     kube.var.log.containers.
        Merge_Log           On
        Merge_Log_Trim      On
        K8S-Logging.Parser  On
        K8S-Logging.Exclude Off
    [INPUT]
        Name              tail
        Path              /var/log/containers/*.log
        Tag               kube.*
        Parser            docker
        DB                /var/log/flb_kube.db
        Refresh_Interval  10
    [OUTPUT]
        Name            es
        Match           *
        Host            es_server
        Port            9200
        Replace_Dots    On
        Index           index
        Buffer_Size     False
        Generate_ID     On
        Trace_Output    Off
        Retry_Limit     False
    [PARSER]
        Name   json
        Format json
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z
    [PARSER]
        Name        docker
        Format      json
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On
```
This is the additional Filter that I'm trying to apply to parse the json under the message field of the kubernetes logs
```
    [FILTER]
        Name                parser
        Match               *
        Key_Name            message
        Parser              json
        Reserve_Data        On
        Unescape_Key        On
        Preserve_Key        On
```

And again this works as well, the message field gets parsed properly except that some of the log messages go missing. Checked the json syntax and it is correct in all of the logs.

In the fluent-bit logs I don't see much other than some failures to send to the Elasticsearch servers, "failed to flush chunk" but I can see those even if the additional filter is applied or not for example
```
failed to flush chunk '1-1601562735.241169178.flb', retry in 10 seconds: task_id=66, input=tail.0 > output=es.0
```
Other things that I can see when setting the log level to debug is
```
KA connection #91 is in a failed state to: es-host, cleaning up
```
Networking is stable so the connection to elastic should be healthy at all times. Even looking through the elasticsearch response outputted in the log I can see that all responses are errors:false. Not sure what I'm doing wrong or what's going on with it.

**Comment by edsiper** [comment]
Hi, 

- your config looks good. But, didn't the Kubernetes log message comes in the ```log``` key and not in a ```message``` key ? 
- if you disable the filter parser, do you get the whole data ?

the KA (keep alive message) is about a persistent connection to elasticsearch, it's normal that gets dropped or in an unused state so Fluent Bit just remove it and create a new one when needed. For short, not an issue

**Comment by alexanderccc** [comment]
Hi @edsiper 

I understand about the KA, so that's not the issue then.

The kubernetes log messages do come in the log key, as I said I'm trying to additionally parse the message key as well since the micro services output in that key a structured json message. 
I'm not sure what you mean in regards to whole data, The data itself isn't corrupted in any way or does it contain any less fields or missing portions in the logs. 

Basically what I'm trying to do with fluent-bit is get the log from kubernetes parse the "log" key as one would normally parse it then also parse the "message" key after parsing the log one, and then send the completely parsed message to elastic. If I just parse the "log" key and not parse the additional message all is fine and works well, it's only when I introduce the additional filter that should parse the "message" key I seem to miss a log or 2 for each micro service. This does happen kind of random. I did consider elasticsearch to be the point of failure but I'm currently sending the data to 3 nodes (quite powerful), I tried sending it directly to it with an elb in front of them or with them configured as endpoints/service in the Kube cluster. Both ways give the same result and elasticsearch is not complaining about any of the incoming data. 

Let me know if you require any additional data, I would really want to continue using fluent-bit standalone and not require any additional service (like Fluentd) to reparse the messages from fluent-bit.
